src/block.rs

```diff
@@ -170,46 +170,114 @@ impl BlockBuilder {
 
 #[cfg(test)]
 mod tests {
+    use rstest::rstest;
+
     use super::*;
-    use crate::test_utils::{gen_attrs, gen_empty_attrs};
+    use crate::test_utils::{assert_debug_snapshot, gen_attrs, gen_empty_attrs};
 
-    #[test]
-    fn test_block() {
+    #[derive(Debug)]
+    struct BlockTestCase {
+        name: &'static str,
+        entries: Vec<(&'static [u8], Option<&'static [u8]>)>, // (key, value) with None for tombstone
+        expected_size: Option<usize>,                         // Expected size of the block
+        use_timestamped_attrs: bool, // Flag to use attributes with a timestamp
+        timestamp: Option<i64>,      // timestamp value
+    }
+
+    fn build_block(test_case: &BlockTestCase) -> Block {
         let mut builder = BlockBuilder::new(4096);
-        assert!(builder.add_value(b"key1", b"value1", gen_empty_attrs()));
-        assert!(builder.add_value(b"key1", b"value1", gen_empty_attrs()));
-        assert!(builder.add_value(b"key2", b"value2", gen_empty_attrs()));
-        let block = builder.build().unwrap();
+
+        for (key, value) in &test_case.entries {
+            let attrs = if test_case.use_timestamped_attrs {
+                gen_attrs(test_case.timestamp.unwrap_or(1)) // Use timestamp value or default to 1
+            } else {
+                gen_empty_attrs()
+            };
+
+            match value {
+                Some(v) => assert!(builder.add_value(key, v, attrs)),
+                None => assert!(builder.add_tombstone(key, gen_empty_attrs())),
+            }
+        }
+        builder.build().expect("Failed to build block")
+    }
+
+    #[rstest]
+    #[case(BlockTestCase {
+        name: "test_block",
+        entries: vec![
+            (b"key1", Some(b"value1")),
+            (b"key1", Some(b"value1")),
+            (b"key2", Some(b"value2")),
+        ],
+        expected_size: None,
+        use_timestamped_attrs: false,
+        timestamp: None,
+    })]
+    fn test_block(#[case] test_case: BlockTestCase) {
```

Comment:
Can you combine all of the test fn's into one? That's kinda the point with these rstests: we should be able to have a  bunch of `BlockTestCase`'s all run against one function. See the `row_codec.rs` tests for an example.

Reply:
@criccomini - I have pushed a change combining the test cases into one. Thanks